src/test/java/api/requests/scenarios/LoanDueDatesAfterRecallTests.java

```diff
@@ -525,6 +526,400 @@ public void changedDueDateAfterRecallingAnItemShouldRespectTenantTimezone()
+    assertThat("due date is the original date",
+        recalledDueDate, not(originalDueDate));
+
+    final String expectedDueDate = ClockManager.getClockManager().getDateTime().plusWeeks(2).toString(ISODateTimeFormat.dateTime());
+    assertThat("due date is not the recall due date (2 weeks)",
+        storedLoan.getString("dueDate"), is(expectedDueDate));
+
+    // Move the fixed clock so that the loan is now overdue
+    ClockManager.getClockManager().setClock(Clock.offset(clock, Duration.ofDays(15)));
+
+    requestsFixture.placeHoldShelfRequest(smallAngryPlanet, charlotte,
+        DateTime.now(DateTimeZone.UTC), requestServicePoint.getId(), "Recall");
+
+    storedLoan = loansStorageClient.getById(loan.getId()).getJson();
+    assertThat("second recall changed the due date (2 weeks)",
+        storedLoan.getString("dueDate"), is(recalledDueDate));
+  }
+
+  @Test
+  public void secondRecallRequestWithMGDTruncationInPlaceDoesNotChangeDueDate()
+      throws InterruptedException,
+      ExecutionException,
+      TimeoutException,
+      MalformedURLException {
+    final IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    final IndividualResource requestServicePoint = servicePointsFixture.cd1();
+    final IndividualResource steve = usersFixture.steve();
+    final IndividualResource jessica = usersFixture.jessica();
+    final IndividualResource charlotte = usersFixture.charlotte();
+
+    final LoanPolicyBuilder canCirculateRollingPolicy = new LoanPolicyBuilder()
+        .withName("Can Circulate Rolling With Recalls")
+        .withDescription("Can circulate item With Recalls")
+        .rolling(Period.weeks(3))
+        .unlimitedRenewals()
+        .renewFromSystemDate()
+        .withRecallsMinimumGuaranteedLoanPeriod(Period.weeks(2))
+        .withRecallsRecallReturnInterval(Period.weeks(1));
+
+    final IndividualResource loanPolicy = loanPoliciesFixture.create(canCirculateRollingPolicy);
+
+    useLoanPolicyAsFallback(loanPolicy.getId(),
+        requestPoliciesFixture.allowAllRequestPolicy().getId(),
+        noticePoliciesFixture.activeNotice().getId());
+
+    final IndividualResource loan = loansFixture.checkOutByBarcode(
+      smallAngryPlanet, steve, ClockManager.getClockManager().getDateTime());
+
+    final String originalDueDate = loan.getJson().getString("dueDate");
+
+    requestsFixture.placeHoldShelfRequest(smallAngryPlanet, jessica,
+        DateTime.now(DateTimeZone.UTC), requestServicePoint.getId(), "Recall");
+
+    JsonObject storedLoan = loansStorageClient.getById(loan.getId()).getJson();
+
+    final String recalledDueDate = storedLoan.getString("dueDate");
+    assertThat("due date is the original date",
+        recalledDueDate, not(originalDueDate));
+
+    final String expectedDueDate = ClockManager.getClockManager().getDateTime().plusWeeks(2).toString(ISODateTimeFormat.dateTime());
+    assertThat("due date is not the recall due date (2 weeks)",
```

Comment:
The check suggests the due date should be the minimum loan period, the message states that it is not.

Am I just getting confused reading the assertion messages, because they state what shouldn't have happened, rather than what should?

Reply:
I think that is the problem here. I agree that it is confusing in the test itself. I need to rethink this or do some more research into what best practices are for unit test failure messages. Should we describe the assertion or the failure of the assertion. In the log you'd see something like "something.something.SomeAssertionError: due date is not the recall due date (2 weeks)". Is that clearer than "something.something.SomeAssertionError: due date is the recall due date (2 weeks)"?

Again, maybe it is better to use a less specific message so that the logs look something like "something.something.SomeAssertionError: due date should be the recall due date (2 weeks)". Or we just drop the messages and let it log something like "something.something.SomeAssertionError: dueDate (some date) not equal to expected expectedDueDate (some other date)" or whatever the default message is.

Reply:
The biggest limitation I've found with the default diagnostic information is that it only has knowledge of the values, not where the values come from (in this case the due date of the loan).

I think I probably write quite vague messages fairly often, e.g. `due date should have changed` or `due date should not have changed`.